src/arcade/core/util/distributions/Distribution.java

```diff
@@ -0,0 +1,52 @@
+package arcade.core.util.distributions;
+
+import ec.util.MersenneTwisterFast;
+import arcade.core.util.MiniBox;
+
+/**
+ * A {@code Distribution} object represents a parameter distribution.
+ *
+ * <p>When created, the object immediately draws an initial value from the distribution. All getters
+ * should return this initial value. Additional values can be pulled from the distribution using the
+ * next methods, but does not change the initial value.
+ *
+ * <p>A new distribution object can be created, in which the initial value is used as the new base
+ * value.
+ */
+public interface Distribution {
+    /**
+     * Gets the initial value drawn from the distribution as a double.
+     *
+     * @return the double drawn from the distribution
+     */
+    double getDoubleValue();
+
+    /**
+     * Gets the initial value drawn from the distribution as an integer.
+     *
+     * @return the integer drawn from the distribution
+     */
+    int getIntValue();
+
+    /**
+     * Gets the distribution parameters.
+     *
+     * @return the distribution parameters
+     */
+    MiniBox getParameters();
+
+    /**
+     * Gets the expected value of the distribution.
+     *
+     * @return the expected value
+     */
+    double getExpected();
+
+    /**
+     * Creates a new distribution based on the initial value drawn.
+     *
+     * @param random the random number generator
+     * @return the new distribution
+     */
+    Distribution rebase(MersenneTwisterFast random);
```

Comment:
I don't love needing to maintain the random number generator explicitly here, is there a use case where random is not a singleton?

Reply:
I'm not sure what the alternative could be, have each `Distribution` keep a copy of `random` when constructed?

Reply:
an alternative is using the `.getNumberGenerator().` from the MASON Distribution abstract class https://cs.gmu.edu/~eclab/projects/mason/classdocs.22/sim/util/distribution/AbstractDistribution.html

that being said, I'm don't have a strong opinion

Reply:
I don't think we have that method in the version of MASON we're using, unfortunately